tests/pytests/unit/modules/test_ansiblegate.py

```diff
@@ -1,200 +1,134 @@
-
-
-def test_resolver_module_loader_failure(resolver):
-    """
-    Test Ansible module loader.
-    :return:
-    """
-    mod = "four.five.six"
-    with pytest.raises(ImportError) as import_error:
-        resolver.load_module(mod)
-
-    mod = "i.even.do.not.exist.at.all"
-    with pytest.raises(LoaderError) as loader_error:
-        resolver.load_module(mod)
+    with patch("subprocess.run") as proc_run_mock:
+        proc_run_mock.return_value.stdout = salt.utils.json.dumps(extension)
+        ret = ansiblegate.help("foo")
+        assert ret["description"] == extension["foo"]["doc"]["description"]
 
 
-def test_resolver_module_loader(resolver):
+def test_virtual_function(subtests):
     """
-    Test Ansible module loader.
+    Test Ansible module __virtual__ when ansible is not installed on the minion.
     :return:
     """
-    with patch("salt.modules.ansiblegate.importlib", MagicMock()), patch(
-        "salt.modules.ansiblegate.importlib.import_module", lambda x: x
-    ):
-        assert resolver.load_module("four.five.six") == "ansible.modules.four.five.six"
 
+    with subtests.test("missing ansible binary"):
+        with patch("salt.utils.path.which", side_effect=[None]):
+            assert ansiblegate.__virtual__() == (
+                False,
+                "The 'ansible' binary was not found.",
+            )
 
-def test_resolver_module_loader_import_failure(resolver):
-    """
-    Test Ansible module loader failure.
-    :return:
-    """
-    with patch("salt.modules.ansiblegate.importlib", MagicMock()), patch(
-        "salt.modules.ansiblegate.importlib.import_module", lambda x: x
-    ):
-        with pytest.raises(LoaderError) as loader_error:
-            resolver.load_module("something.strange")
+    with subtests.test("missing ansible-doc binary"):
+        with patch(
+            "salt.utils.path.which", side_effect=["/path/to/ansible", None],
+        ):
+            assert ansiblegate.__virtual__() == (
+                False,
+                "The 'ansible-doc' binary was not found.",
+            )
 
+    with subtests.test("missing ansible-playbook binary"):
+        with patch(
+            "salt.utils.path.which",
+            side_effect=["/path/to/ansible", "/path/to/ansible-doc", None],
+        ):
+            assert ansiblegate.__virtual__() == (
+                False,
+                "The 'ansible-playbook' binary was not found.",
+            )
 
-def test_virtual_function(resolver):
-    """
-    Test Ansible module __virtual__ when ansible is not installed on the minion.
-    :return:
-    """
-    with patch("salt.modules.ansiblegate.ansible", None):
-        assert ansible.__virtual__() == "ansible"
+    with subtests.test("missing ansible-playbook binary"):
+        with patch(
+            "salt.utils.path.which",
+            side_effect=[
+                "/path/to/ansible",
+                "/path/to/ansible-doc",
+                "/path/to/ansible-playbook",
```

Comment:
**question**: is this a misnamed subtest? Says missing playbook, but it's right here in the patch 🤔